src/mesh/FloodingRouter.cpp

```diff
@@ -60,6 +60,10 @@ void FloodingRouter::sniffReceived(const meshtastic_MeshPacket *p, const meshtas
         }
     }
 
+    if (config.device.rebroadcast_mode == meshtastic_Config_DeviceConfig_RebroadcastMode_LOCAL_ONLY) {
```

Comment:
I think `LOCAL_ONLY` isn’t working correctly like this. Firstly, it will only end up at the clause when it didn’t go to `Router::send()` before. Secondly, cancelSending is only doing something when there is already a packet with that ID in the queue (in the first clause at line 34 a rebroadcast is canceled when in the meantime it already heard an ACK). Right now I think you will see 

> “cancelSending id=xxx, removed=0”

meaning nothing is canceled. 

I think we should just set `encryptedOk` in the RoutingModule to false to ensure we don’t rebroadcast encrypted packets. Then, if you configure channels on a repeater with a different PSK than the default, we already also ignore packets from ‘open’ networks, right?